### ISSUE TITLE
Retrieval models - specify flag for not retrieving ground truth reply 

### ISSUE THREAD

**Issue by jazzminewang** [facebookresearch/ParlAI#1697]
Retrieval models (eg. kvmemnn) on personachat training split frequently retrieve the ground truth. A solution to this would be adding a flag for the model to retrieve the second-ranked response when the first-ranked response is the ground truth. 

Alternatively, adding a flag to display the weighting of each response so that users could create this logic manually. I see some scores displayed but it's unclear if they can be used for this purpose. 

``` [label_candidates: my mom was single with 3 boys , so we never left the projects .|i try to wear all black every day . it makes me feel comfortable .|well nursing stresses you out so i wish luck with sister|yeah just want to pick up nba nfl getting old|i really like celine dion . what about you ?|...and 15 more]
   [candidate_scores]: [329.3330652  329.3330652  329.3330652  329.2982666  329.24543604]```

in the output for the default TFIDF model when set to eval mode, but this does not appear to be part of the output for other retrieval models, and also does not match the # of total label candidates.

**Comment by jazzminewang** [comment]
I'm specifically hoping to do this with the pretrained model from kvmemnn (https://github.com/facebookresearch/ParlAI/blob/master/projects/convai2/baselines/kvmemnn/eval_f1.py)  as well as the internal pretrained transformer ranker model. 

I'm aware of the possibility of subclassing torch ranker, but not sure how this would apply to pretrained models.
https://github.com/facebookresearch/ParlAI/blob/master/docs/source/tutorial_torch_ranker_agent.rst#creating-a-model

**Comment by jazzminewang** [comment]
So I just took a closer look at this and posting for posteriority in case others are confused/this is incorrect.

1. This specifies model file. https://github.com/facebookresearch/ParlAI/blob/master/projects/convai2/baselines/kvmemnn/eval_f1.py#L17

2. Going to the model file, in this case (https://github.com/facebookresearch/ParlAI/blob/master/projects/personachat/kvmemnn/kvmemnn.py), and modify the retrieval logic directly.
